### PR TITLE
vespa-cli 7.462.20 (new formula)

### PR DESCRIPTION
**Pull request by mpolden**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This is the official command-line tool for working with Vespa (https://vespa.ai). It supports working with local Vespa installations as well as the Vespa Cloud service.

Regarding the explicit `version` field: The tags in https://github.com/vespa-engine/vespa/ are not ideal. I think they're RPM package versions, where the trailing `-1` is epoch. However, the official version number should not include this.

I could've done:
```
version "7.462.20"
url "https://github.com/vespa-engine/vespa/archive/refs/tags/vespa-#{version}-1.tar.gz"
 ```

but then `brew audit` complains.

FYI @bratseth

### CODE REVIEW

File: Formula/vespa-cli.rb
Comment (carlocab) [inline Formula/vespa-cli.rb:20]:
```suggestion
      (bash_completion/"vespa").write Utils.safe_popen_read(bin/"vespa", "completion", "bash")
      (fish_completion/"vespa.fish").write Utils.safe_popen_read(bin/"vespa", "completion", "fish")
      (zsh_completion/"_vespa").write Utils.safe_popen_read(bin/"vespa", "completion", "zsh")
```

File: Formula/vespa-cli.rb
Comment (carlocab) [inline Formula/vespa-cli.rb:26]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Comment (mpolden) [reply Formula/vespa-cli.rb:26]: The problem is that this is just a client. You need to have a running Vespa installation for it to do something useful.

Comment (mpolden) [reply Formula/vespa-cli.rb:26]: I suppose I can make it change a config option (`vespa config`) and read it back, would that be sufficient?

Comment (carlocab) [reply Formula/vespa-cli.rb:26]: Can we try to access a non-existent server and check we get the expected error?

Checking the config would be good too.

Comment (mpolden) [reply Formula/vespa-cli.rb]: PTAL

File: Formula/vespa-cli.rb
Comment (chenrui333) [inline Formula/vespa-cli.rb:4]: same artifact, but shorter url
```suggestion
  url "https://github.com/vespa-engine/vespa/archive/vespa-7.462.20-1.tar.gz"
```

File: Formula/vespa-cli.rb
Comment (chenrui333) [inline Formula/vespa-cli.rb:25]:
```suggestion
    ENV["VESPA_CLI_HOME"] = testpath
```

**Review by carlocab — DISMISSED** [review]
Thanks, @mpolden.

**Comment by mpolden** [comment]
Anything else needed here before merging?

**Comment by carlocab** [comment]
No, I was just waiting for CI to finish running. @BrewTestBot will merge this shortly.

**Comment by carlocab** [comment]
Oops. @BrewTestBot doesn't like that your PR branch is also called `master`. (More specifically, trying to query the GitHub API for the associated workflow run when your branch is called `master` returns a 502.)

That's easy enough to fix. Leave it to me.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**
